### PR TITLE
Adjust ROSA-HCP 4.14 job

### PR DESCRIPTION
**Pull request by LiangquanLi930**
Synchronize jobs from version 4.15 to 4.14 and delete redundant jobs.

### CODE REVIEW

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml [hunk 1]
```diff
@@ -994,140 +994,46 @@ tests:
...
     cluster_profile: aws-sd-qe
     env:
       CHANNEL_GROUP: nightly
       E2E_RUN_TAGS: '@amd64 and @aws-ipi and @rosa and @hypershift-hosted'
-      OPENSHIFT_VERSION: "4.14"
-      TEST_FILTERS: ~ChkUpgrade&;~NonPreRelease&;~Serial&;~Disruptive&;~DisconnectedOnly&;~HyperShiftMGMT&;~MicroShiftOnly&;~NonHyperShiftHOST&;ROSA&
-    test:
-    - chain: openshift-e2e-test-hypershift-qe
-    workflow: rosa-aws-sts-hypershift
-- as: aws-rosa-sts-hcp-advanced-full-f14
```
Comment (yasun1) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:1009]: Let us keep the same job list as 4.15. On the stable.yaml, we only do the cluster health checking, while do the OCP e2e testing with the nightly builds to cover the EC/RC and the Z stream openshift version.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-stable.yaml [hunk 2]
```diff
@@ -258,41 +258,12 @@ tests:
...
+  cron: 29 10 5,14,21,28 * *
   steps:
     cluster_profile: aws-sd-qe
     env:
       CHANNEL_GROUP: candidate
-      OPENSHIFT_VERSION: "4.14"
-    test:
-    - chain: cucushift-installer-check-cluster-health
-    workflow: rosa-aws-sts-hypershift
-- as: aws-rosa-sts-hcp-advanced-full-f14
```
Comment (yasun1) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-stable.yaml:271]: Let us keep the same job list as 4.15. In the stable channel, currently, we need to run all of the workflows on the staging OCM env to do the cluster health checking with the 4.14.z openshift version.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml (hunk 1 above)
Comment (yasun1) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:1009]: These jobs are not redundant jobs, we need to keep them here.

Comment (LiangquanLi930) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:1009]: > Let us keep the same job list as 4.15

Yes, I just copied the 4.15 jobs to 4.14. There were too many jobs before, so we deleted some jobs

**Comment by LiangquanLi930** [comment]
@liangxia Could you help to review this PR? Thanks

**Pull request merged**